gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/tpgm/diff/DiffTest.java

```diff
@@ -40,18 +40,49 @@ public void testDiffOnGraph() throws Exception {
       "__valFrom: 1543700000000L, _diff: 1})" +
       "(carole:Person {name : \"Carol\", gender : \"f\", city : \"Dresden\", age : 30," +
       "__valFrom: 1543600000000L, _diff: 0})" +
-      "(gpse)-[:hasModerator {since : 2013 , __valFrom : 1543700000000L, __valTo : " +
-      "1543800000000L, _diff: 0}]->(davee)" +
       "(gpse)-[:hasMember{ __valFrom: 1543600000000L, __valTo: 1543800000000L, _diff: -1}]->" +
       "(davee) (gpse)-[:hasMember {_diff: 0}]->(carole)" +
       "-[:knows {since : 2014 , __valFrom : 1543700000000L, _diff: 1}]->(davee)" +
       "]");
     LogicalGraph inputGraphEpgm = loader.getLogicalGraphByVariable("g3");
-    TemporalGraph temporalGraph = inputGraphEpgm.toTemporalGraph(
-      GradoopFlinkTestBase::extractGraphHeadTime, GradoopFlinkTestBase::extractVertexTime,
-      GradoopFlinkTestBase::extractEdgeTime);
+    TemporalGraph temporalGraph = toTemporalGraph(inputGraphEpgm);
     TemporalGraph result = temporalGraph.diff(new AsOf(1543600000000L), new AsOf(1543800000000L));
     collectAndAssertTrue(loader.getLogicalGraphByVariable("expected").equalsByData(
       result.toLogicalGraph()));
   }
+
+  /**
+   * Test the temporal diff operator with validation.
+   *
+   * @throws Exception when the execution in Flink fails.
+   */
+  @Test
+  public void testDiffOnGraphWithValidate() throws Exception {
```

Comment:
Please have a look at it here ... set valid to false makes no difference ... maybe the equals by data does not recognize the invalid edge